### PR TITLE
Added BrickID as third input parameter

### PR DESCRIPTION
**Pull request by GiuliaPaggi**
Can be called from emureco_*/RUN*/ with 3 parameters: from_plate to_plate BrickID (defined by wall number in the tens position and brick number for the units, eg W5B1->51) instead of having to change the file.

### CODE REVIEW

File: run1analysis/recoparameters/linkingloop.sh [hunk 1]
```diff
@@ -1,38 +1,54 @@
...
 cp firstlink.rootrc link.rootrc
 
-makescanset -set=11.0.0.0 -dzbase=195 -from_plate=$1 -to_plate=$2 -v=2
+makescanset -set=$3.0.0.0 -dzbase=195 -from_plate=$1 -to_plate=$2 -v=2
 
-emlink -set=11.0.0.0 -new -v=2
+emlink -set=$3.0.0.0 -new -v=2
 
-cp b000011.0.0.0.link.ps plot_prelink/b000011.0.0.0.prelink_$1_$2.ps
+cp b0000$3.0.0.0.link.ps plot_prelink/b0000$3.0.0.0.prelink_$1_$2.ps
```
Comment (antonioiuliano2) [inline run1analysis/recoparameters/linkingloop.sh]: We need to generalize this line to 1-digit and 3-digit brickIDs

File: run1analysis/recoparameters/linkingloop.sh [hunk 2]
```diff
@@ -1,38 +1,54 @@
...
 cp secondlink.rootrc link.rootrc
 
-makescanset -set=11.0.0.0 -dzbase=195 -from_plate=$1 -to_plate=$2 -v=2
+makescanset -set=$3.0.0.0 -dzbase=195 -from_plate=$1 -to_plate=$2 -v=2
 
-emlink -set=11.0.0.0 -new -v=2
+emlink -set=$3.0.0.0 -new -v=2
 
-cp b000011.0.0.0.link.ps plot_link/b000011.0.0.0.link_$1_$2.ps
+cp b0000$3.0.0.0.link.ps plot_link/b0000$3.0.0.0.link_$1_$2.ps
```
Comment (antonioiuliano2) [inline run1analysis/recoparameters/linkingloop.sh]: Same as above

File: run1analysis/recoparameters/linkingloop.sh (hunk 1 above)
Comment (GiuliaPaggi) [reply run1analysis/recoparameters/linkingloop.sh]: Ok I'm trying. I get the 3 digit, but can't understand the 1 digit brickID, in which situation would you have it

Comment (antonioiuliano2) [reply run1analysis/recoparameters/linkingloop.sh]: It is better to make the code general if we can, FEDRA can have brickIDs up to 6 digits (the reason for b000001). Of course, in our case it is not needed to go that far.

**Comment by GiuliaPaggi** [comment]
Hi Antonio, is something still missing in my second commit you want me to add?

**Comment by antonioiuliano2** [comment]
Hello Giulia, Thank you for this commit.

I was just thinking that it may be more practical and compact to use printf syntax for the 6-digit padding, instead of all the if conditions. For example in a simple bash terminal:

printf "b%0*d\n" 6 11 

returns

b000011

I wanted to test it in this script, but I did not have time to test it yet sorry

**Comment by GiuliaPaggi** [comment]
I did not know that exist, it simplifies a lot and works for any number of digits now

**Comment by antonioiuliano2** [comment]
Perfect, thank you. I have just another question: I see you have added the cd ./$workdir command at the beginning.

Does not this mean the code is not expected to be launched in the brick folder anymore, but in the folder above it? Just to know to avoid confusion and mistakes when I will launch it in my folders

**Comment by GiuliaPaggi** [comment]
Yes, I thought that adding the brick ID as parameter one can have just the one .sh in the RUN folder and run it from there, is there any problems I'm not thinking about in doing that?

**Comment by antonioiuliano2** [comment]
No, it is okay. You also wrote it in the comment at the beginning, so it is clear. Also you check the folder exists, so there should not be room for error. For me it is all ok now

**Comment by antonioiuliano2** [comment]
Thank you again for this update :) !

**Pull request merged**